src/audio/drc/Testbench_drc_sin/drc_sin_fixed.c

```diff
@@ -0,0 +1,610 @@
+	1929888719,
+	1932769410,
+	1935631909,
+	1938476189,
+	1941302224,
+	1944109986,
+	1946899450,
+	1949670588,
+	1952423376,
+	1955157787,
+	1957873795,
+	1960571374,
+	1963250500,
+	1965911147,
+	1968553291,
+	1971176905,
+	1973781966,
+	1976368449,
+	1978936330,
+	1981485584,
+	1984016188,
+	1986528117,
+	1989021349,
+	1991495859,
+	1993951624,
+	1996388621,
+	1998806828,
+	2001206221,
+	2003586778,
+	2005948477,
+	2008291295,
+	2010615209,
+	2012920200,
+	2015206244,
+	2017473320,
+	2019721407,
+	2021950483,
+	2024160528,
+	2026351521,
+	2028523441,
+	2030676268,
+	2032809981,
+	2034924561,
+	2037019987,
+	2039096240,
+	2041153301,
+	2043191149,
+	2045209766,
+	2047209132,
+	2049189230,
+	2051150040,
+	2053091543,
+	2055013722,
+	2056916559,
+	2058800035,
+	2060664132,
+	2062508835,
+	2064334123,
+	2066139982,
+	2067926393,
+	2069693341,
+	2071440807,
+	2073168776,
+	2074877232,
+	2076566159,
+	2078235539,
+	2079885359,
+	2081515602,
+	2083126253,
+	2084717297,
+	2086288719,
+	2087840504,
+	2089372637,
+	2090885104,
+	2092377891,
+	2093850984,
+	2095304369,
+	2096738031,
+	2098151959,
+	2099546138,
+	2100920555,
+	2102275198,
+	2103610053,
+	2104925108,
+	2106220351,
+	2107495769,
+	2108751351,
+	2109987084,
+	2111202958,
+	2112398959,
+	2113575079,
+	2114731304,
+	2115867625,
+	2116984030,
+	2118080510,
+	2119157053,
+	2120213650,
+	2121250291,
+	2122266966,
+	2123263665,
+	2124240379,
+	2125197099,
+	2126133816,
+	2127050521,
+	2127947205,
+	2128823861,
+	2129680479,
+	2130517051,
+	2131333571,
+	2132130029,
+	2132906419,
+	2133662733,
+	2134398965,
+	2135115106,
+	2135811152,
+	2136487094,
+	2137142926,
+	2137778643,
+	2138394239,
+	2138989707,
+	2139565042,
+	2140120239,
+	2140655292,
+	2141170196,
+	2141664947,
+	2142139540,
+	2142593970,
+	2143028233,
+	2143442325,
+	2143836243,
+	2144209981,
+	2144563538,
+	2144896909,
+	2145210091,
+	2145503082,
+	2145775879,
+	2146028479,
+	2146260880,
+	2146473079,
+	2146665075,
+	2146836865,
+	2146988449,
+	2147119824,
+	2147230990,
+	2147321945,
+	2147392689,
+	2147443221,
+	2147473541,
+	2147483647
+};
+
+/*
+ * Input is Q2.30: (-2.0, 2.0)
+ * Output range: (-1.0, 1.0); regulated to Q1.31: (-1.0, 1.0)
+ */
+int main()
+{
+	int32_t x[TEST_VECTOR];
+	int8_t i;
+	mkdir("Results", 0777);
+	FILE* fd = fopen("Results/drc_sin_fixed.txt", "w");
+	fprintf(fd, " %10s  %10s %10s \n", "idx", "in-sine", "out-sine");
+	init_sinedata_fixpt(x);
+	for (i = 0; i < TEST_VECTOR; i++)
+	{
+		fprintf(fd, " %10d %11li ", i, x[i]);
+		drc_sin_fixed(x[i], i, fd);
+	}
+	fclose(fd);
+}
+
+
+/*degree is input in Q2.30 */
+/*radian is outputin Q1.31 */
+inline int32_t drc_sin_fixed(int32_t x, int8_t i, FILE* fd)
+{
+	const int32_t PI_OVER_TWO = Q_CONVERT_FLOAT(1.57079632679489661923f, 30);
+	fprintf(fd, " %13li\n", sin_fixed(q_mult(x, PI_OVER_TWO, 30, 30, 28)));
+	return sin_fixed(q_mult(x, PI_OVER_TWO, 30, 30, 28));
+}
+
+/* Compute fixed point sine with table lookup and interpolation */
+int32_t sin_fixed(int32_t w)
```

Comment:
should we directly include the original implementation in src/math?

Reply:
https://github.com/thesofproject/sof/pull/3715#discussion_r549822913
Yes-I would like so, Curtis has some ideas as described in comments.